### ISSUE TITLE
Can driverless list printers via _ipps._tcp

### ISSUE THREAD

**Issue by tangyanli** [OpenPrinting/cups-filters#251]
Hi,

Is there some method to set driverless lists the network printers via _ipps._tcp?

PS:
I can use ippfind to list the network printers via _ipps._tcp as below.
$ ippfind _ipps._tcp

**Comment by nidhijainpnp** [comment]
Hi @tangyanli ,


> PS:
> I can use ippfind to list the network printers via _ipps._tcp as below.
> $ ippfind _ipps._tcp

Yes,it is right that ,you can list the network printers and show their uri by `ippfind`.But,I am unable to understand your issue.Can you please elaborate more,what actually you want to do /or the error you are facing?

**Comment by tillkamppeter** [comment]
The difference between `ippfind` and `driverless` is that the former lists all IPP printers, including remote CUPS queues and IPP printers which do not support driverless printing. `driverless` uses `ipptool` but calls it with filter options, exactly showing only printers which support at least IPP version 2.0 and at least one of the Page Description Languages (PDLs) PDF, Apple Raster, PWG Raster, and PCLm.
In addition, `ippfind` shows URIs in standard IPP format whereas `driverless` shows URIs in DNS-SD-service-name-based format.
@tangyanli suggests to add support for IPPS-based (encrypted) IPP print services.
My suggestion is, by default to show each printer once, if it appears with both IPP and IPPS, show the IPPS version. Also introduce command line options to show only IPP entries or only IPPS entries.

**Comment by tangyanli** [comment]
@nidhijainpnp 
The issue I faced is: 
If I set IPP->TLS off at the printer side. I can find the printer at the web page: [localhost:631]->[Administration]->[Add Printer].
![1](https://user-images.githubusercontent.com/54799082/87491331-b9ed7980-c5fc-11ea-83bd-30724ebc4b27.png)
But if I set IPP->TLS on at the printer side. I can't find the printer at above web page any more.

@tillkamppeter Thanks for your explanation and suggestion. And I think your suggestion can properly resolved this issue.
